doc/source/relationships.rst

```diff
@@ -70,6 +70,111 @@ Relationship Uniqueness
+        airplane they can operate
+        """
+        airplane = neomodel.StringProperty(required = True)
+        
+This means that either of these concrete objects can appear at the end 
+of a ``friends_with`` relationship and be instantiated to the right object.
+
+Here is a minimal example to demonstrate that::
+
+    # Create some technical persons
+    A = TechnicalPerson(name = "Grumpy", expertise = "Grumpiness").save()
+    B = TechnicalPerson(name = "Happy", expertise = "Unicorns"}).save()
+    C = TechnicalPerson(name = "Sleepy", expertise = "Pillows"}).save()
+    
+    # Create some Pilot Persons
+    D = PilotPerson(name = "Porco Rosso", airplane = "Savoia-Marchetti").save()
+    E = PilotPerson(name = "Jack Dalton", airplane = "Beechcraft Model 18").save()
+    
+    # TechnicalPersons befriend Technical Persons
+    A.friends_with.connect(B)
+    B.friends_with.connect(C)
+    C.friends_with.connect(A)
+    
+    # Pilot Persons befriend Pilot Persons
+    D.friends_with.connect(E)
+    
+    # Technical Persons befriend Pilot Persons
+    A.friends_with.connect(D)
+    E.friends_with.connect(C)
+    
+    for some_friend in A.friends_with:
+        print(some_friend)
+        
+This will show two friends connected with node "Grumpy", one of which is a ``TechnicalPerson`` 
+and the other a ``PilotPerson``.
+
+
+Automatic class resolution
+--------------------------
+
+Neomodel is able to transform nodes to objects, automatically, via a *node-class registry* 
+that is progreessively built up during the definition of the models.
+
+The dictionary provides a mapping from the set of labels associated with a node to the class 
+that is implied by this set of labels. In the example above, the *node-class registry* would contain 
+the following entries:
+
+    * BasePerson                  --> BasePerson
+    * BasePerson, TechnicalPerson --> TechnicalPerson
+    * BasePerson, PilotPerson     --> PilotPerson
+    
+This automatic resolution is effected by ``neomodel.Database.cypher_objectaware_query`` which is 
+invoked automatically where needed but it can also be invoked in exactly the same way as ``neomodel.
+Database.cypher_query``, manually, wherever this functionality may be needed.
+
+Doing so however, requires a bit of caution. As a consequence of the way the *node-class registry* is 
+built up and used, if ``BasePerson``, ``TechnicalPerson``, ``PilotPerson`` were defined 
+in separate files / modules and one of them was not loaded prior to a query to the database, then the 
+*node-class registry* would be unaware of the labels that leads to that particular class and would raise 
+``neomodel.exception.ModelDefinitionMismatch``.
+
+If the exception is not handled, it produces an error message that returns the labels of the node that 
+was retrieved from the database as well as the current *node-class registry*. These two pieces of 
+information can be used to debug the model mismatch further.
```

Comment:
if i understand correctly, the use of 'mismatch' seems unappropriate in the designated context. shouldn't it be rather have a notion of 'undefined', 'unknown' or 'not found'?

Reply:
"...seems unappropriate" --> "...seems inappropriate..."

The model mismatch refers to the condition of the OGM encountering a node that bears more or less labels than what its definition implies.

For example, in the hierarchy "Pilot extends Person", nodes of type "Pilot" are expected to bear the labels "Pilot" and "Person".

If a node was found that contained "Pilot, Person, FighterPilot" then that would be a mismatch with the definitions of the model.